rust-runtime/aws-smithy-types/src/body/http_body_1_x.rs

```diff
@@ -23,6 +23,50 @@ impl SdkBody {
     {
         SdkBody::from_body_0_4_internal(Http1toHttp04::new(body.map_err(Into::into)))
     }
+
+    /// Construct an `SdkBody` from a type that implements [`hyper_1_0::body::Body<Data = Bytes>`](hyper_1_0::body::Body)
+    pub fn from_hyper_1_x<T, E>(body: T) -> Self
+    where
+        T: hyper_1_0::body::Body<Data = Bytes, Error = E> + Send + Sync + 'static,
+        E: Into<Error> + 'static,
+    {
+        SdkBody {
+            bytes_contents: None,
+            inner: super::Inner::Dyn {
+                inner: super::BoxBody::HttpBody10(http_body_util::combinators::BoxBody::new(
+                    body.map_err(Into::into),
+                )),
+            },
+            rebuild: None,
+        }
+    }
+}
```

Comment:
how does this type differ from `HttpBody`? aren't they kinda the same ? I don't think we necessarily want a hyper-specific constructor.